Refactor/dependency injection

gordominossi opened the pull request.

gordominossi committed:
Changing file directory structure to be more like that of template-node

gordominossi committed:
Adding dependency injection for the createUser mutation

gordominossi committed:
Adding dependency injection for the user and users mutations

gordominossi committed:
Adding dependency injection for the login mutation

gordominossi committed:
Locking typedi dependency version

gordominossi committed:
Adding container to GraphQL schema
Using inject and constructors instead of Container.get()

alanraso commented on src/business/rule/create-user.use-case.ts:
```diff
@@ -0,0 +1,42 @@
+
+@Service()
+export class CreateUserUseCase extends UseCase<UserInputModel, UserModel> {
+  @Inject()
+  protected readonly dataSource: UserDbDataSource;
```
Dá pra fazer o mesmo tipo de injeção q vc fez nos use-cases la noresolver: pelos construtors

alanraso commented on src/business/rule/use-case.ts:
```diff
@@ -0,0 +1,6 @@
+import { DataSource } from '@src/data/source/datasource';
+
+export abstract class UseCase<TRequest, TResult> {
+  protected abstract readonly dataSource: DataSource<any>;
+  abstract exec(requestObject: TRequest): Promise<TResult>;
+}
```
Dado que essa abstração não tem nada de muito concreto, acho que podera ser só um a interface né. O que acha?

gordominossi replied:
Sim.
É que eu não queria deixa público esse dataSource

alanraso commented on src/data/database-setup.ts:
```diff
@@ -0,0 +1,19 @@
+import { Container } from 'typedi';
+import { createConnection, useContainer } from 'typeorm';
+
+if (process.env.NODE_ENV === 'test') {
+  dotenv.config({ path: `${process.cwd()}/.env.test` });
+} else {
+  dotenv.config({ path: `${process.cwd()}/.env` });
+}
```
Não tem como deixar em um lugar só centralizado essa configuração? 🤔

gordominossi replied:
🤔

alanraso reviewed (approved):
🚀

gordominossi committed:
Changing env setup to occur only once

gordominossi committed:
Changing directory structure,
Increasing hierachy depth and improving generic types,
Changing DI to constructors instead of @Inject()s
